### ISSUE TITLE
Minor: 3.0 Bug/Question: duplicate records in hooks.

### ISSUE THREAD

**Issue by ShakAttacked** [orientechnologies/orientdb#8041]
### OS: Centos

Hi,

as part of my prototype, I've implemented ODatabaseLifecycleListener to see what happens. In the life cycle, I was registering the hook (I also have it in the xml config) (see below).

I also have onrecordAfterCreate where I add a new vertex to User for each record created on Person. (not a real example but for test purpose).  I noticed that my logs in onOpen was called very often, it was printing a dozen times and then it would pause for a bit and then start again.

the issue I have is that when I insert a record in Person, the hook is triggered (hook is only registering for Person changes) but onRecordAfterCreate gets called 3 times instead of just once.  
when I remove the LifeCycleListener, there is no problem.  is this a bug? can you clarify when to use the lifecycle listener. I thought I could create multiple hooks and potentially register them here instead of the config file but now I'm not sure how and if it's possible.

basically, I have 
 <hooks>
        <hook class="kinbox.hooks.MyHook" position="REGULAR"/>
    </hooks>

and 

   @Override
    public void onOpen(ODatabaseInternal oDatabaseInternal)
    {
        System.out.println(">>>>>>>>>>>>>>>>>>>>>>>>>>>>onOpen");
        oDatabaseInternal.registerHook(this);

    }

Regards,
Safwan
